pkg/usersignup/usersignup.go

```diff
@@ -30,5 +35,12 @@ func TransformUsername(username string) string {
 	if strings.HasSuffix(newUsername, "-") {
 		newUsername = newUsername + "crt"
 	}
+	if len(newUsername) > MaxLength {
+		newUsername = newUsername[0:MaxLength]
+		if !dnsRegexp.MatchString(newUsername) {
+			// trim starting or ending hyphen
+			newUsername = strings.Trim(newUsername, "-")
```

Comment:
Do we know why we append `crt` as a prefix on line 33 and on a suffix on line 36? Maybe it should just be trimmed instead of appending `crt`?

Btw, I'm not suggesting we remove the trim here if we do that. I recognize that we'd still need it here since we shorten it to 20 chars and the last character could end up being `-`. So it's more a question of whether that `crt` stuff still has value or whether the reason for doing that would have an impact on what we should do here with regard to trimming the `-`.

Reply:
crt doesn't have any value on its own as far as I remember. So, I guess trimming would be fine.